src/react-components/avatar-selector.js

```diff
@@ -0,0 +1,119 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { injectIntl, FormattedMessage } from 'react-intl';
+
+class AvatarSelector extends Component {
+  static propTypes = {
+    avatars: PropTypes.array,
+    avatar: PropTypes.string,
+    onChange: PropTypes.func,
+  }
+
+  getAvatarIndex(direction=0) {
+    const currAvatarIndex = this.props.avatars.findIndex(avatar => avatar.id === this.props.avatar);
+    const numAvatars = this.props.avatars.length;
+    return ((currAvatarIndex + direction) % numAvatars + numAvatars) % numAvatars;
+  }
+
+  nextAvatar = () => {
+    const newAvatarIndex = this.getAvatarIndex(1);
+    this.props.onChange(this.props.avatars[newAvatarIndex].id);
+  }
+
+  prevAvatar = () => {
+    const newAvatarIndex = this.getAvatarIndex(-1);
+    this.props.onChange(this.props.avatars[newAvatarIndex].id);
+  }
+
+  componentDidUpdate() {
+    // HACK - a-animation ought to restart the animation when the `to` attribute changes, but it doesn't
+    // so we need to force it here.
+    const currRot = this.animation.parentNode.getAttribute('rotation');
```

Comment:
You probably want to filter for specific component updates here otherwise if we add more properties we could end up restarting the animation unintentionally.